spark2/src/main/java/org/apache/iceberg/spark/source/CustomCatalogs.java

```diff
@@ -40,6 +43,12 @@
 
 public final class CustomCatalogs {
   private static final Cache<Pair<SparkSession, String>, Catalog> CATALOG_CACHE = Caffeine.newBuilder()
+      .expireAfterAccess(10, TimeUnit.MINUTES)
```

Comment:
Should this be configurable?

Reply:
I'm worried this will cause the same issues we had here https://github.com/apache/iceberg/pull/1674

Reply:
I am aware of #1674. This is slightly different from #1674, the catalog is bounded to spark session.  I'm thinking whether this can be configurable so that the user could set an expiration strategy according to different scenarios. While it is neither not a table property nor a spark session property.

Reply:
While it is bound to `SparkSession`, it seems introducing a cache timeout can still cause the issues Russell mentioned above.

cc @rymurr @rdblue who developed these places

Reply:
I agree, I think the timed expiration is potentially worse here. If we expire by time then we will be evicting a catalog that is definitely still being used by a `SparkSession`. Whereas before it was _maybe_ being used. 

I think that #2325 may actually be the right solution. It moves to a cross catalog pool and handles reclaiming client connections regardless of the underlying catalog. Then any catalogs in this cache will have 0 connections attached to them and can stay in the cache forever. @lcspinter do you agree?

Reply:
Great! I checked the PR, it looks very promising! Thanks @RussellSpitzer @aokolnychyi @rymurr!